### PR TITLE
Add subtest for autotest.pm

### PR DESCRIPTION
**Pull request by gauravpathak**
- add subtest in 08-autotest.t for start_process

### CODE REVIEW

File: t/08-autotest.t [hunk 1]
```diff
@@ -468,6 +468,26 @@ subtest 'test skipping tests' => sub {
...
 };
 
+subtest 'start_process' => sub {
+    my $fh;
+    open($fh, '<', \"fake-file");    # the file handle will be closed automatically inside start_process
+    $autotest::isotovideo = $fh;
+    # Test the start_process subroutine
+    my ($process, $child) = autotest::start_process();
+
+    # Assertions
```
Comment (perlpunk) [inline t/08-autotest.t:478]:
```suggestion
```
`is`, `isa_ok`, `ok` etc. are all assertions; I think that is nothing that needs to be documented.

File: t/08-autotest.t [hunk 2]
```diff
@@ -468,6 +468,26 @@ subtest 'test skipping tests' => sub {
     stderr_like { autotest::runalltests } qr/skipping/, 'Skipping Test Run';
 };
 
+subtest 'start_process' => sub {
+    my $fh;
+    open($fh, '<', \"fake-file");    # the file handle will be closed automatically inside start_process
```
Comment (perlpunk) [inline t/08-autotest.t:473]:
```suggestion
    open $fh, '<', \"fake-file";
```
The fact that variables go out of scope at the end of the block is something that we shouldn't have to comment everywhere.

File: t/08-autotest.t [hunk 3]
```diff
@@ -468,6 +468,25 @@ subtest 'test skipping tests' => sub {
...
+    my $fh;
+    open $fh, '<', \"fake-file";
+    $autotest::isotovideo = $fh;
+    # Test the start_process subroutine
+    my ($process, $child) = autotest::start_process();
+
+    isa_ok $process, 'HASH', 'start_process returns a HASH reference';
+    isa_ok $child, 'GLOB', '$child is a GLOB reference';
+
+    # Check if autoflush was called on both handles
```
Comment (perlpunk) [inline t/08-autotest.t:481]:
```suggestion
```
That's already what the test labels below are saying

File: t/08-autotest.t [hunk 4]
```diff
@@ -468,6 +468,25 @@ subtest 'test skipping tests' => sub {
...
+    ok $child->autoflush, 'autoflush was called on child handle';
+    ok $fh->autoflush, 'autoflush was called on isotovideo handle';
+
+    open($fh, '<', \"fake-file");    # the file handle will be closed automatically inside start_process
```
Comment (perlpunk) [inline t/08-autotest.t:485]:
```suggestion
    open $fh, '<', \"fake-file";
```

File: autotest.pm [hunk 5]
```diff
@@ -313,9 +313,9 @@ sub start_process () {
             $0 = "$0: autotest";
             my $line = <$isotovideo>;
             if (!$line) {
-                _exit(0);
+                _exit(0);    # uncoverable statement
             }
-            print "GOT $line\n";
+            bmwqemu::diag("GOT $line\n");
```
Comment (okurz) [inline autotest.pm:318]: I agree with the change but it shouldn't be hidden in the commit "Add subtest for autotest.pm". Please make that a separate commit

**Review by okurz — CHANGES_REQUESTED** [review]

File: autotest.pm (hunk 5 above)
Comment (gauravpathak) [reply autotest.pm:318]: Pushed this change in a new commit

**Review by okurz — APPROVED** [review]

**Review by Wabri — APPROVED** [review]
LGTM
